### PR TITLE
#4999 soft delete vaccine course item

### LINKED ISSUE #4999: Can't initialise because of missing VaccineCourseItem
## What went wrong? 😲
I added some vaccine courses in an OMS central server and tried to initialised a remote site, but get this error message.
photo_2024-10-04 14 33 47

## Expected behaviour 🤔
Should be able to initialise with correct data

## How to Reproduce 🔨

Steps to reproduce the behaviour:

## Your environment 🌱

- Open mSupply Version: 2.3.0-RC5
- Legacy mSupply Central Server Version:

- Platform: browser central and android remote

- Database type:

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #4999

# 👩🏻‍💻 What does this PR do?

Adds soft deletes for vaccine course items. Remote sites can now be initialised! 🙈 

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Create a vaccine course, with at least one vaccine item - save
- [ ] Reopen vaccine course, remove the vaccine item - save
- [ ] Initialise a remote site
- [ ] Should initialise fine 🚀 

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

**Review by roxy-dao — COMMENTED** [review]
Creating a lot of changelogs for 1 change 👀 and have to press sync twice to get OMS central server changes 👀

**Comment by lache-melvin** [comment]
Thanks Roxy, have updated so only upsert items if they're actually new (we can't "update" vaccine course items so i think it is ok)

I'm getting the changes after only one sync? 👀

File: server/service/src/vaccine_course/update.rs [hunk 1]
```diff
@@ -212,7 +227,42 @@ pub fn generate(
...
-    Ok((updated_course, doses_to_delete))
+    let items_for_course = VaccineCourseItemRepository::new(&connection).query_by_filter(
+        VaccineCourseItemFilter::new().vaccine_course_id(EqualFilter::equal_to(&id)),
+    )?;
+
+    // Should remove any items that are not in the new list
+    let vaccine_items_to_delete = items_for_course
+        .clone()
+        .into_iter()
+        .filter_map(|item| {
```
Comment (roxy-dao) [inline server/service/src/vaccine_course/update.rs:238]: 🔥

**Review by roxy-dao — APPROVED** [review]
Thanks Lache~

I'm still having to press twice to sync... but 🤷🏻

**Pull request merged**